autowarefoundation/autoware
# Adding nvidia-docker v2 support

## Status

**PRODUCTION**, safe for the "optional" todos mentioned below.

## Description

Adding nvidia docker v2 support as discussed in issue autowarefoundation/autoware_ai#93 .

## Todos

> I'd like to leave these up to the Autoware maintainers.

- [ ] I think these new images should become the standard (as Autoware assumes or even needs an nvidia GPU to run on, right?) You can consider renaming current images to something like "-legacy", rather than suffixing these new images with "-nvidia".
- [ ] Documentation could be checked on all old references to the old `nvidia-docker run` command (which is `docker run --runtime nvidia` these days).

## Steps to Test or Reproduce

See autowarefoundation/autoware_ai#93.

Make sure to install nvidia-docker v2 and see if the old image works for your (should not work. Opening RVIZ would crash for example). Then try again with the new image and see if it works.
